_posts/tutorials/2021-05-20-frontends-with-javalinmithril.md

```diff
@@ -0,0 +1,450 @@
+
+I came across JavalinVue a year or so ago, I absolutely loved the simplicity, so much so that I have contributed to it,  and the performance and quick development time have been key to the success of my projects and initatives(also a story for another time). This Plugin is inspired by JavalinVue(and a dare), and adresses the only shortcoming I found in it : you need to manage your component ids to be unique, otherwise it doesnt work correctly.
+
+In this tutorial, we will go over the main workflow in JavalinMithril, and explain some choices along the way. There is no build system for the front end, But there is a well defined namespace(package) and class system that we need to stick to to get the results we want. 
+
+## Setup
+Our backend will be Java, and we'll be using Maven to build.
+We need to bring in Javalin (web library), Jackson (JSON serializer), and JavalinMithril (front end). We also need the Webjar for Mithril.js\\
+We'll also add Vue (view library) for our frontend:
+
+```markup
+<dependency>
+    <groupId>io.javalin</groupId>
+        <artifactId>javalin</artifactId>
+        <version>3.13.6</version
+        <type>jar</type>
+    </dependency>
+    <dependency>
+        <groupId>io.github.javalin</groupId>
+        <artifactId>javalinmithril</artifactId>
+        <version>0.1.0</version>
+         <type>jar</type>
+    </dependency>
+    <dependency>
+        <groupId>org.webjars.npm</groupId>
+        <artifactId>mithril</artifactId>
+        <version>2.0.4</version>
+    </dependency>
+    <dependency>
+        <groupId>com.fasterxml.jackson.core</groupId>
+        <artifactId>jackson-databind</artifactId>
+        <version>2.10.3</version>
+    </dependency>
+```
+
+<div class="comment" markdown="1">
+You can add all frontend dependencies as [Webjars](https://www.webjars.org/), which can be built directly from NPM.
+If something is available on NPM, it's also available as a Webjar, but as a prebuilt dist version (it has no dependencies).
+To view the full POM, please go to [GitHub](https://github.com/TareqK/javalinmithril-example/blob/master/pom.xml).
+</div>
+
+Now that we have all our dependencies in order, we need to configure our web server. Lets start off with 2 classes : the Main class, and the app class.
+
+Starting off, our Main class, in `/src/main/java/io/javalinmithril/demo/Main.java` is just a wrapper class(which I add because I am a sad non-kotlin boi :C  and I like neat class names too much C: ) that calls and starts our App
+
+```java
+import io.javalin.Javalin
+package io.javalin.javalinmithril.demo;
+
+public class Main {
+    
+    public static void main(String[] args){
+        new App().start();
+    }
+}
+```
+
+Our App class, which has the start method, looks like this
+
+```java
+...
```

Comment:
I wouldn't omit the class/imports, but up to you.

Reply:
It's a verbosity thing they are there in the code